ext/drbbdup/drbbdup.c

```diff
@@ -288,13 +288,20 @@ drbbdup_add_copy(void *drcontext, instrlist_t *bb, instrlist_t *orig_bb)
 static drbbdup_manager_t *
 drbbdup_create_manager(void *drcontext, void *tag, instrlist_t *bb)
 {
-    drbbdup_manager_t *manager = dr_global_alloc(sizeof(drbbdup_manager_t));
+    /* This per-block memory can add up: for 2.5M basic blocks we can take up >512M
+     * of space, which if it's in the limited-size vmcode region is a problem.
+     * We thus explicitly request unreachable heap.
+     * XXX: Maybe DR should break compatibility and change the default.
+     */
+    drbbdup_manager_t *manager = dr_custom_alloc(
```

Comment:
Why don't you pass drcontext here?

Reply:
I did in the first iteration but that API routine actually yells at you if you don't pass NULL for global heap.

Reply:
Nit: Maybe add a comment to indicate we want to use the global heap.

Reply:
Done.